demo/popcha-client/web/src/main.py

```diff
@@ -0,0 +1,222 @@
+"""
+This files includes the basis of the example server. Here are managed all
+HTTP/S requests.
+"""
+import json
+from os import path
+import secrets
+import urllib.parse
+from typing import IO
+
+from flask import Flask, redirect, request, Response, render_template
+from flask_wtf import CSRFProtect
+
+from src.counterApp import CounterApp
+from src.authentication import Authentication
+
+# Define the global variables
+config: dict = {}
+core_app: CounterApp
+authenticationProvider: Authentication
+app = Flask("Example_authentication_server",
+            template_folder = path.join(path.dirname(__file__), "templates")
+            )
+
+
+def check_config(config_file: IO) -> bool:
```

Comment:
I would try to limit the number of places where you access global variables. It makes functions easier to understand. Here for instance you could accept a dictionary object, modify it and return the modified dictionary object. The caller then updated the global variable / write the data back to disk.

Also I think it would not hurt logging some message saying "Hey I updated your config file"